django/db/backends/postgresql/introspection.py

```diff
@@ -138,7 +139,15 @@ def get_key_columns(self, cursor, table_name):
                 ON ccu.constraint_catalog = tc.constraint_catalog
                     AND ccu.constraint_schema = tc.constraint_schema
                     AND ccu.constraint_name = tc.constraint_name
-            WHERE kcu.table_name = %s AND tc.constraint_type = 'FOREIGN KEY'
+            JOIN pg_catalog.pg_class cl
+                ON cl.relname = kcu.table_name
+            JOIN pg_catalog.pg_namespace n
+                ON cl.relnamespace = n.oid
+                    AND n.nspname = kcu.table_schema
+            WHERE kcu.table_name = %s
+                AND tc.constraint_type = 'FOREIGN KEY'
+                AND cl.relkind IN ('r', 'v')
+                AND pg_catalog.pg_table_is_visible(cl.oid)
```

Comment:
Is all of this necessary?, why not just:
```SQL
JOIN pg_class c ON c.relname = kcu.table_name
WHERE
    kcu.table_name = %s AND
    tc.constraint_type = 'FOREIGN KEY' AND
    pg_catalog.pg_table_is_visible(c.oid)
```
You don't need to add `pg_catalog.`

Reply:
1. Without INNER JOIN of `pg_catalog` we are not filtering by schema name.

2. I'm removing `pg_catalog.` prefix.

3. It seems that the query from `get_relations` returns the same data. I was thinking of reusing it, but it seemed too intrusive for this PR.

Reply:
I was confused why we have two different queries in `get_key_columns()` and `get_relations()` that return the same columns but in a different format, so I unified them in #11577. What do you think? it should make this change much simpler.

Reply:
Looks good to me. If you merge it, I can rebase this PR on top.

Reply:
Merged.